src/codecs/mod.rs

```diff
@@ -0,0 +1,18 @@
+#[cfg(test)]
```

Comment:
Would these two trees best start out their life in `lib/vector-core` instead of in `src` proper? I expect we will want to move them there eventually.

Hmm, I see that this is using `config::DataType` still, which hasn't been moved into the core lib. @blt was that on your agenda to move later?

Reply:
Hmm, since we might implement a large series of framing / codec options, and these not being required by default, I'm not sure if they should be moved into `vector-core`.

Reply:
However, a `vector-codecs` lib would be nice to implement those in isolation.

Reply:
I've actually worked pretty hard to keep configuration notions out of core. My thinking has been that core should be generic pieces that you can use to make a vector but isn't, itself, vector. Doing that makes testing for correctness and performance of an individual core component less entangled with the rest of the project. Config strikes less a generic notion and more something very specific to vector-the-binary.